CMakeLists.txt

```diff
@@ -3,21 +3,21 @@ cmake_minimum_required(VERSION 3.0)
 project(ssb-dbgen # SSB benchmark data (and query) generator utilities
 	LANGUAGES C)
 
+set(CMAKE_C_FLAGS_DEBUG "-g")
+set(CMAKE_C_FLAGS_RELEASE "-O3")
+
 set(DATABASE "DB2" CACHE STRING "Command language dialect to target with the qgen query generator (\"DATABASE\"")
 set_property(CACHE DATABASE PROPERTY STRINGS INFORMIX DB2 TDAT SQLSERVER SYBASE)
 
-set(MACHINE LINUX CACHE STRING "Platform to target (\"MACHINE\")")
-set_property(CACHE MACHINE PROPERTY STRINGS ATT HP IBM ICL MVS SGI SUN U2200 VMS LINUX WIN32 MAC)
-
+set(MACHINE_DOCSTRING "Platform to target (\"MACHINE\")")
 if (WIN32)
-	set(MACHINE "WIN32")
-endif()
-if (UNIX AND NOT APPLE)
-	set(MACHINE "LINUX")
-endif()
-if (APPLE)
-	set(MACHINE "MAC")
+	set(MACHINE WIN32 CACHE STRING ${MACHINE_DOCSTRING})
+elseif (UNIX AND NOT APPLE)
+	set(MACHINE LINUX CACHE STRING ${MACHINE_DOCSTRING})
+elseif (APPLE)
+	set(MACHINE MAC CACHE STRING ${MACHINE_DOCSTRING})
```

Comment:
I don't quite understand what's going on here. 

Now, I know it's actually not a good idea to have the user specify a machine string at all, and we should just detect the actual features, but given that this _is_ specified manually - why these changes?

... oh, I see, you're relying on CMake's internal variables. Interesting. However, I'm only willing to have that done when the user has not specifically set a value for `MACHINE`, to more closely follow original behavior. Also remember that not all Unix flavors are either Linux or Mac; for example - Sun, SGI and AT&T Unix.